.github/workflows/docs-ingestion.yml

```diff
@@ -0,0 +1,33 @@
+name: Docs website data ingestion to Algolia for semantic search
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - packages/apps/docs/**
+
+jobs:
+  ingestion:
+    name: Ingestion docs website data to Algolia
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # 👈 Required to retrieve git history, needed to determine diffs.
```

Comment:
Do you really need fetch depth 0? This means all history is included, while you likely only need the latest state?

Reply:
yeah, only needs the latest state